### PR TITLE
(DOCSP-26638): new create, read, update, and delete pages

### PR DESCRIPTION
**Pull request by mongomoe**
### Jira

- https://jira.mongodb.org/browse/DOCSP-26638

### Staged Changes

- [CRUD - Create - React Native SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26638-CRUD-docs/sdk/react-native/realm-database/crud/create/)
- [CRUD - Read - React Native SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26638-CRUD-docs/sdk/react-native/realm-database/crud/read/)
- [CRUD - Update - React Native SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26638-CRUD-docs/sdk/react-native/realm-database/crud/update/)
- [CRUD - Delete - React Native SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26638-CRUD-docs/sdk/react-native/realm-database/crud/delete/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [N/A] Create Jira ticket for corresponding docs-app-services update(s), if any
- [N/A] Checked/updated Admin API
- [N/A] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/realm-database/crud/create.txt [hunk 1]
```diff
@@ -13,22 +13,54 @@ CRUD - Create - React Native SDK
...
-and then pass it to :js-sdk:`Realm.create() <Realm.html#create>` inside of a
-write transaction. If the realm's :ref:`schema <react-native-realm-schema>` includes
-the object type and the object conforms to the schema, then Realm
-stores the object, which is now *managed* by the realm.
-   
-.. literalinclude:: /examples/generated/node/read-and-write-data.snippet.read-and-write-create-a-new-object.js
-  :language: javascript
-  :emphasize-lines: 6
\ No newline at end of file
+To add an object to a realm, instantiate it as you would any other object and
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/create.txt:37]: I'm not totally clear on what "instantiate it as you would any other object" is referring to. Does this mean the realm or the object? Maybe we could make this a bit clearer?

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 2]
```diff
@@ -103,11 +141,38 @@ guarantees a consistent order of results if you explicitly sort them.
 To sort a query, call the :js-sdk:`sorted() <Realm.Collection.html#sorted>`
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/read.txt:141]: Again, can't comment on line 136, but I think we should introduce `useQuery()` earlier. See my suggestion for the previous section.

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 3]
```diff
@@ -103,11 +141,38 @@ guarantees a consistent order of results if you explicitly sort them.
...
-  :emphasize-lines: 4, 6, 8, 13
-
-To sort a query based on a property of an :ref:`embedded object
-<react-native-embedded-objects>` or a :ref:`related object
-<react-native-client-relationships>`, use dot-notation as if it were in a
-regular, nested object.
\ No newline at end of file
+In the following example of a ``TaskList`` component, we use the ``useQuery()``
+hook to initially retrieve the set of Task objects. We then use the
+``sorted()`` method to:
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/read.txt:146]:
```suggestion
``sorted()`` method to work with the data in various ways:
```

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 4]
```diff
@@ -103,11 +141,38 @@ guarantees a consistent order of results if you explicitly sort them.
...
+
+#. Sort objects based on the task's name alphabetically.
+#. Sort objects based on the task's name alphabetically in descending order.
+#. Sort objects based on the task's priority in descending order and the task's name in ascending order.
+#. Sort objects based on the assignee object's name alphabetically.
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/read.txt:151]: I don't think order is important here, as they're all separate. Let's make this an unordered list.
```suggestion
- Sort objects based on the task's name alphabetically.
- Sort objects based on the task's name alphabetically in descending order.
- Sort objects based on the task's priority in descending order and the task's name in ascending order.
- Sort objects based on the assignee object's name alphabetically.
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 5]
```diff
@@ -14,24 +14,54 @@ About the Examples on this Page
 -------------------------------
 The examples on this page use the following schemas:
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:15]:
```suggestion
The examples on this page use the following schema:
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 6]
```diff
@@ -14,24 +14,54 @@ About the Examples on this Page
...
 
 Update an Object
 ----------------
 
-You can add, modify, or delete properties of a Realm object inside
-of a write transaction in the same way that you would update any other
-JavaScript object.
+You can add, modify, or delete properties of a Realm object inside of a
+:ref:`write transaction <react-native-write-transactions>` in the same way that
+you would update any other JavaScript object.
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:38]: Minor suggestion for clarity.
```suggestion
You can add, modify, or delete properties of a Realm object in the same way that
you would update any other JavaScript object. But, you must do it inside of a
:ref:`write transaction <react-native-write-transactions>`.
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 7]
```diff
@@ -14,24 +14,54 @@ About the Examples on this Page
...
+
+In the following example of a ``TaskItem`` component, we:
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:40]:
```suggestion
To update an object, we have to do a few things:
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 8]
```diff
@@ -14,24 +14,54 @@ About the Examples on this Page
...
+
+#. Get access to the opened realm instance by calling the ``useRealm()`` hook within the component. 
+#. Retrieve a task by calling ``useObject()` with "Task" and the ``_id`` parameter of the component.
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:43]:
```suggestion
#. Retrieve a task by calling ``useObject()`` with "Task" and the ``_id`` parameter of the component.
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 9]
```diff
@@ -46,14 +76,42 @@ JavaScript object.
...
+
+   While you can create new :js-sdk:`Realm Objects <Realm.Object.html>` by
+   calling the :mdn:`new <Web/JavaScript/Reference/Operators/new>` operator
+   within a write transaction, you **must** call ``Realm.create()`` to upsert an
+   object.
 
-.. literalinclude:: /examples/generated/node/read-and-write-data.snippet.read-and-write-upsert-an-object.js
-  :language: javascript
-  :emphasize-lines: 4, 13
+In the following example of a ``CreateTaskItem`` component we:
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:91]:
```suggestion
To upsert an object, we need to do a few things:
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 10]
```diff
@@ -46,14 +76,42 @@ JavaScript object.
...
+<Realm.html#create>` with the update mode set to ``modified``. The operation
+either inserts a new object with the given primary key or updates an existing
+object that already has that primary key.
+
+.. note:: Upserting Requires ``Realm.create()``
+
+   While you can create new :js-sdk:`Realm Objects <Realm.Object.html>` by
+   calling the :mdn:`new <Web/JavaScript/Reference/Operators/new>` operator
+   within a write transaction, you **must** call ``Realm.create()`` to upsert an
+   object.
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:89]: Suggestion for clarity. It's good to put the critical info first.

```suggestion
   You **must** call ``Realm.create()`` within a write transaction to upsert an object. 
   This is different than creating a new :js-sdk:`Realm Objects <Realm.Object.html>` by
   calling the :mdn:`new <Web/JavaScript/Reference/Operators/new>` operator.
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 11]
```diff
@@ -46,14 +76,42 @@ JavaScript object.
...
 
-.. literalinclude:: /examples/generated/node/read-and-write-data.snippet.read-and-write-upsert-an-object.js
-  :language: javascript
-  :emphasize-lines: 4, 13
+In the following example of a ``CreateTaskItem`` component we:
+
+#. Get access to the opened realm instance by calling the ``useRealm()`` hook within the component. 
+#. Perform a write transaction, and :ref:`create <react-native-create-a-new-object>` a ``Task`` object with an ``_id`` value of ``1234``.
+#. Call `` Realm.create()`` inside the write transaction to upsert a ``Task`` object by specifying the same ``_id`` and a different``progressMinutes`` and the update mode set to "modified".
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:95]:
```suggestion
#. Call ``Realm.create()`` inside the write transaction to upsert a ``Task`` object by specifying the same ``_id`` and a different``progressMinutes`` and the update mode set to "modified".
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 12]
```diff
@@ -63,14 +121,29 @@ Bulk Update a Collection
...
-  :emphasize-lines: 7, 9
-
-.. note:: Inverse Relationships
-
-   Thanks to an :ref:`inverse relationship <react-native-inverse-relationship>` from
-   ``Dog.owner`` to ``Person.dogs``, Realm Database automatically updates
-   Ali's list of dogs whenever we set her as a puppy's owner.
+<Web/JavaScript/Reference/Statements/for...of>`). In the loop, update each object individually.
+
+In the following example of a ``TaskDashboard`` component, we:
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:126]:
```suggestion
To bulk update a collection, we need to do a few things:
```

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 13]
```diff
@@ -63,14 +121,29 @@ Bulk Update a Collection
...
+
+#. Get access to the opened realm instance by calling the ``useRealm()`` hook within the component.
+#. Retrieve all tasks in the realm instance by passing "Task" to the ``useQuery()`` hook.
+#. Create a component method resetProgressOnAllTasks() that performs a write transaction. Within that write transaction, we bulk update all tasks by looping through them using ``for...of`` and set their ``progressMinutes`` to 0.
```
Comment (krollins-mdb) [inline source/sdk/react-native/realm-database/crud/update.txt:130]:
```suggestion
#. Create a component method ``resetProgressOnAllTasks()`` that performs a write transaction. Within that write transaction, we bulk update all tasks by looping through them using ``for...of`` and set their ``progressMinutes`` to 0.
```

**Review by krollins-mdb — COMMENTED** [review]
This is looking great, @mongomoe! I have some suggestions, mostly for concision and clarity.

I noticed you used a specific construction to introduce code samples, like this:
> In the following example of a ``TaskDashboard`` component, we:

I suggested changing these throughout for a couple reasons:
- Using incomplete introductory sentences for lists can be hard to parse, especially for non-native English readers.
- I don't think we need to specify that the examples are components, but if we need that info, we should probably include it int he examples themselves. The introductory text is so far removed on the page that it doesn't feel directly connected to the examples.

File: source/sdk/react-native/realm-database/crud/delete.txt [hunk 14]
```diff
@@ -30,9 +38,31 @@ Delete an Object
 To delete an object from a realm, pass the object to :js-sdk:`Realm.delete()
 <Realm.html#delete>` inside of a write transaction.
 
-.. literalinclude:: /examples/generated/node/read-and-write-data.snippet.read-and-write-delete-object.js
-  :language: javascript
-  :emphasize-lines: 3
+In the following example of a ``DogList`` component, we:
+
+#. Get access to the opened realm instance by calling the ``useRealm()`` hook.
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/delete.txt:43]: Closing this and all related "Get access to the opened realm" vs "Open a realm instance" comments since we discussed this on slack (how "useRealm()" doesn't actually open the realm, etc).

**Comment by mongomoe** [comment]
Fyi @krollins-mdb removed any similar/repeating comments per our discussion, comments such as:
-  "Get access to the opened realm" vs "Open a realm instance" 
- Unordered vs Ordered list

File: source/sdk/react-native/realm-database/crud/create.txt [hunk 15]
```diff
@@ -13,22 +13,54 @@ CRUD - Create - React Native SDK
...
+then use the :mdn:`new <Web/JavaScript/Reference/Operators/new>` operator to
+create a new realm object inside of a write transaction. If the realm's
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/create.txt:39]: Altered the paragraph a bit, let me know if it is good.

File: source/sdk/react-native/realm-database/crud/create.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/create.txt]: To instantiate an object in JavaScript you use the new operator, like so you could do something like
```
class MyClass {
}
let m1 = new MyClass();
```

With the new Realm JS Class Based Models updates, we can instantiate a Realm Objects the same way, with the new operator (as long as its within a write transaction)

```
class MyRealmClass extends Realm.Object {
    static schema = { ...schema }
}
realm.write(() => {
    let m1 = new MyRealmClass(realm, {...myProperties})
})

```

[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/crud/create.txt]: Right. Sorry, perhaps I could've been clearer. The pronoun "it" is ambiguous. Could refer to the object you're adding to the realm or the realm itself. So, should this read
> instantiate the object you want to add as you would any other object

or

> instantiate the realm as you would any other object

[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/create.txt]: Oh that makes sense, so I've altered the text now to be:
```
"To add an object to a realm instance, instantiate the :js-sdk:`Realm Object
<Realm.Object.html>` as you would any other object by using the :mdn:`new
<Web/JavaScript/Reference/Operators/new>` operator to create a new Realm object
inside of a write transaction. "
```

File: source/sdk/react-native/realm-database/crud/create.txt [hunk 16]
```diff
@@ -13,22 +13,54 @@ CRUD - Create - React Native SDK
...
+:ref:`schema <react-native-realm-schema>` includes the object type and the
+object conforms to the schema, then Realm stores the object, which is now
+*managed* by the realm.
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/create.txt]: A managed object refers to a JavaScript object that has been added to a Realm. 

So:
 
```
 let myDog = new Dog(realm, {name: dogName, age: 1});
```
is just a regular ole JS object. 

But when you do:

```
realm.write(() => {
 let myDog = new Dog(realm, {name: dogName, age: 1});
 });
```
"myDog" becomes "managed" by Realm. 


More concisely, a managed object refers to data in the realm, and a unmanaged object is just a JavaScript object that hasn't been added to a realm.

File: source/sdk/react-native/realm-database/crud/create.txt [hunk 17]
```diff
@@ -13,22 +13,54 @@ CRUD - Create - React Native SDK
...
+#. Get access to an open realm instance by calling the ``useRealm()`` hook within the component. 
+#. Create a component method ``handleAddItem()`` that performs a :ref:`write transaction <react-native-write-transactions>`, and creates a new ``Dog`` :js-sdk:`object <Realm.Object.html>` based on the TextInput value.
+#. Add an `onPress <https://reactnative.dev/docs/handling-touches>`__ event on the submit button that calls ``handleAddItem()`` 
+
+.. tabs-realm-languages::
+      
+   .. tab::
+      :tabid: typescript
+      
+      .. literalinclude:: /examples/generated/react-native/ts/create-test.snippet.crud-create-object.tsx
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/create.txt:55]: I kinda like having both properties (name and age). Having age shows users that you create a value for an optional field in the same way (by just creating a  key value pair: `age: 1`). We could add an input where the app user inserts a value for age, but I don't think it's necessary, and just makes the code snippet larger than necessary (while achieving the same goal).

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 18]
```diff
@@ -12,10 +12,27 @@ CRUD - Read - React Native SDK
...
+      :tabid: javascript
+
+      .. literalinclude:: /examples/generated/react-native/js/Person.snippet.js-person-schema.js
+         :language: javascript
+
+      .. literalinclude:: /examples/generated/react-native/js/Task.snippet.js-task-schema.js
+         :language: typescript
 
 
 .. _react-native-read-operations:
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/read.txt:38]: Great suggestion! I've added an intro sentence that is a slight variation to your suggestion:

```
Read operations are queries to find your data stored in Realm Database. Use the
``@realm/react`` hooks to find specific objects and get collections by object
type.
```

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 19]
```diff
@@ -28,54 +45,75 @@ Read Operations
...
-  :language: javascript
+   .. tab::
+      :tabid: javascript
+
+      .. literalinclude:: /examples/generated/react-native/js/read-test.snippet.crud-read-filter-data.jsx
+         :language: javascript
+         :emphasize-lines: 2-3, 5-11
+         :linenos:
 
 .. tip:: Filter on Related and Embedded Object Properties
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/read.txt]: Whoops accidentally took this suggestion and realized I added it to the wrong line. Fixed now so that it'll be the subsection title of the sort section.

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 20]
```diff
@@ -28,54 +45,75 @@ Read Operations
...
 
-Filter Queries
-~~~~~~~~~~~~~~
+Query an Object Type and Filter Queries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Query operations return a collection of Realm objects that match the query as a
+:js-sdk:`Realm.Results <Realm.Results.html>` object. A basic query matches all
+objects of a given type in a realm, but you can also apply a filter to the
+collection to find specific objects.
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/read.txt:87]: This is kind of a stylistic choice but I've been trying to avoid combining the Realm paragraphs with the @realm/react explanations. The way I'm mostly trying to structure these examples is:

```
First, explain the concept in Realm Database (for example, here we explain what a query operation returns) 

Then, explain how the component utilizes @realm/react  hooks in the following code example.
```

Basically, we should explain the core concepts of Realm Database first, and then explain how we're using @realm/react to achieve them (since fundamentally, @realm/react is a convenience library. Although obviously, in some areas @realm/react will need more explaining, such as how it handles notifications/reacting to changes automatically/under the hood). 

Regarding, the previous section "Find a Specific Object by Primary Key", the main reason "useObject" is mentioned earlier is because there's not as much explaining that needs to be done about finding an object by its primary key.

File: source/sdk/react-native/realm-database/crud/read.txt (hunk 2 above)
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/read.txt:145]: Similar thoughts to https://github.com/mongodb/docs-realm/pull/2398#discussion_r1052662383

File: source/sdk/react-native/realm-database/crud/create.txt (hunk 16 above)
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/crud/create.txt]: Yep, I get that. But readers may not understand that nuance. We haven't really said what "managed" means. That's why I asked if we might have a good place to link off to. Or we could provide some more context here. In my experience, a lot of devs don't like "magic" things to handle their code and want to understand what's happening.

**Review by krollins-mdb — APPROVED** [review]
Thanks for making changes, @mongomoe! I think there's still one thing worth looking at again (I added [a new comment](https://github.com/mongodb/docs-realm/pull/2398#discussion_r1057705299)), but it's not a blocker.

File: source/sdk/react-native/realm-database/crud/create.txt (hunk 16 above)
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/react-native/realm-database/crud/create.txt]: I just decided to get rid of the "which is now *managed* by the realm." part since the term is potentially confusing and it is basically redundant (since a managed realm object is just an object stored in a realm, which the previous sentence fragment states).

File: source/sdk/react-native/realm-database/crud/update.txt [hunk 21]
```diff
@@ -63,14 +120,29 @@ Bulk Update a Collection
...
-
-   Thanks to an :ref:`inverse relationship <react-native-inverse-relationship>` from
-   ``Dog.owner`` to ``Person.dogs``, Realm Database automatically updates
-   Ali's list of dogs whenever we set her as a puppy's owner.
+<Web/JavaScript/Reference/Statements/for...of>`). In the loop, update each object individually.
+
+In the following example of a ``TaskDashboard`` component, we:
+
+#. Get access to the opened realm instance by calling the ``useRealm()`` hook within the component.
+#. Retrieve all tasks in the realm instance by passing "Task" to the ``useQuery()`` hook.
```
Comment (takameyer) [inline source/sdk/react-native/realm-database/crud/update.txt:128]:
```suggestion
#. Retrieve all tasks in the realm instance by passing ``Task`` to the ``useQuery()`` hook.
```

**Review by takameyer — COMMENTED** [review]
I have couple suggestions
